sam/models/tests/test_lasso_model.py

```diff
@@ -0,0 +1,95 @@
+import os
+import random
+
+import numpy as np
+import pytest
+from sam.feature_engineering.simple_feature_engineering import SimpleFeatureEngineer
+from sam.models import LassoTimeseriesRegressor
+from sam.models.tests.utils import (
+    assert_get_actual,
+    assert_performance,
+    assert_prediction,
+    get_dataset,
+)
+from sklearn.preprocessing import StandardScaler
+
+# If tensorflow is not available, skip these unittests
+skipkeras = False
+try:
+    import tensorflow as tf
+except ImportError:
+    skipkeras = True
+
+
+@pytest.mark.skipif(skipkeras, reason="Keras backend not found")
+@pytest.mark.parametrize(
+    "predict_ahead,quantiles,average_type,use_diff_of_y,y_scaler,max_mae",
+    [
+        pytest.param((0,), (), "mean", True, None, 3, marks=pytest.mark.xfail),  # should fail
+        pytest.param((0, 1), (), "mean", True, None, 3, marks=pytest.mark.xfail),  # should fail
+        ((0,), (), "mean", False, None, 3),  # plain regression
+        ((0,), (0.1, 0.9), "mean", False, None, 3.0),  # quantile regression
+        ((0,), (), "median", False, None, 3.0),  # median prediction
+        ((0,), (), "mean", False, StandardScaler(), 3.0),  # scaler
+        ((1,), (), "mean", False, None, 3.0),  # forecast
+        ((1,), (), "mean", True, None, 3.0),  # use_diff_of_y
+        ((1, 2, 3), (), "mean", False, None, 3.0),  # multiforecast
+        ((1, 2, 3), (), "mean", True, None, 3.0),  # multiforecast with use_diff_of_y
+        ((0,), (0.1, 0.5, 0.9), "mean", False, None, 3.0),  # quantiles
+        ((1, 2, 3), (0.1, 0.5, 0.9), "mean", False, None, 3.0),  # quantiles multiforecast
+        ((1,), (), "mean", True, StandardScaler(), 3.0),  # all options except quantiles
+        ((1, 2, 3), (0.1, 0.5, 0.9), "mean", True, StandardScaler(), 3.0),  # all options
+    ],
+)
+def test_lasso(
+    predict_ahead,
+    quantiles,
+    average_type,
+    use_diff_of_y,
+    y_scaler,
+    max_mae,
+):
+
+    # Now start setting the RNG so we get reproducible results
```

Comment:
Maybe we can make a decorator for this to control the randomness for all our unit tests, instead of defining this for the unit tests of every other model.

Reply:
Please review added decorator :)